cmd/celestia-appd/cmd/root.go

```diff
@@ -129,20 +129,9 @@ func initRootCommand(rootCommand *cobra.Command, capp *app.App) {
 
 	modifyRootCommand(rootCommand)
 
-	// find start command
-	startCmd, _, err := rootCommand.Find([]string{"start"})
-	if err != nil {
-		panic(fmt.Errorf("failed to find start command: %w", err))
-	}
-	startCmdRunE := startCmd.RunE
-
-	// Add the BBR check to the start command
-	startCmd.RunE = func(cmd *cobra.Command, args []string) error {
-		if err := checkBBR(cmd); err != nil {
-			return err
-		}
-
-		return startCmdRunE(cmd, args)
+	// Add hooks run prior to the start command
+	if err := addPreStartHooks(rootCommand, checkAndUpdateMinGasPrices, checkBBR); err != nil {
```

Comment:
Thanks for the refactor, seems cleaner than it was before. 

I was going to suggest adding these pre start hooks to the `start` command directly but unfortunately the start command gets added to the root command via code in Cosmos SDK so we'd have to make the modification in celestiaorg/cosmos-sdk which isn't as nice as making the change in celestia-app.

The other [extremely optional] idea is to copy the start command from Cosmos SDK to celestia-app and then add the PreStartHooks directly in that file (e.g. `start.go`) but this has other downsides.

Ref: https://github.com/celestiaorg/cosmos-sdk/blob/5259747ebf054c2148032202d946945a2d1896c7/server/start.go#L147

Reply:
Yeah there was a time I wanted to bring the start (and init) commands over to celestia-app not just for this but to also make the start up function consistent in our integration tests